configure.ac

```diff
@@ -0,0 +1,420 @@
+AC_ARG_ENABLE([cbcp],
+    AS_HELP_STRING([--enable-cbcp], [Enable Callback Protocol]))
+AM_CONDITIONAL(WITH_CBCP, test "x${enable_cbcp}" = "xyes")
+AM_COND_IF([WITH_CBCP],
+    AC_DEFINE([CBCP_SUPPORT], 1, [Have Callback Protocol support]))
+
+#
+# Disable support for IPX control protocol
+AC_ARG_ENABLE([ipxcp],
+    AS_HELP_STRING([--disable-ipxcp], [Disable IPX Control Protocol support]))
+AM_CONDITIONAL(WITH_IPXCP, test "x${enable_ipxcp}" != "xno")
+AM_COND_IF(WITH_IPXCP,
+    AC_DEFINE([IPX_CHANGE], 1, ["Have IPX Control Protocol"]))
+AS_IF([test "x${build_sunos}" = "xyes" && test "x${enable_ipxcp}" != "xno"],
+    [AC_MSG_ERROR([IPXCP is not supported on SunOS, disable using --disable-ipxcp])])
+
+#
+# Disable support for limiting session duration by maximum octets
+AC_ARG_ENABLE([maxoctets],
+    AS_HELP_STRING([--disable-maxoctets], [Disable support for limiting session by maximum octets]))
+AS_IF([test "x$enable_maxoctets" != "xno"],
+    AC_DEFINE([MAXOCTETS], 1, ["Limit sessions by maximum number of octets"]))
+
+#
+# Disable Microsoft extensions will remove CHAP and MPPE support
+AC_ARG_ENABLE([microsoft-extensions],
+    AS_HELP_STRING([--disable-microsoft-extensions], [Disable Microsoft CHAP / MPPE extensions]))
+
+AM_CONDITIONAL(WITH_CHAPMS, test "x${enable_microsoft_extensions}" != "xno")
+AM_COND_IF([WITH_CHAPMS],
+    AC_DEFINE([CHAPMS], 1, ["Have Microsoft CHAP support"]))
+
+AM_CONDITIONAL(WITH_MPPE, test "x${enable_microsoft_extensions}" != "xno")
+AM_COND_IF([WITH_MPPE],
+    AC_DEFINE([MPPE], 1, ["Have Microsoft MPPE support"]))
+
+#
+# Enable Microsoft LAN Manager support, depends on Microsoft Extensions
+AC_ARG_ENABLE([mslanman],
+    AS_HELP_STRING([--enable-mslanman], [Enable Microsoft LAN Manager support]))
+AS_IF([test "x${enable_mslanman}" = "xyes" && test "x${enable_microsoft_extensions}" != "xno"],
+    AC_DEFINE([MSLANMAN], 1, ["Have Microsoft LAN Manager support"]))
+
+#
+# Disable IPv6 support
+AC_ARG_ENABLE([ipv6-support],
+    AS_HELP_STRING([--disable-ipv6-support], [Disable IPv6 support]))
+AM_CONDITIONAL(WITH_INET6, test "x${enable_ipv6_support}" != "xno")
+AM_COND_IF([WITH_INET6],
+    AC_DEFINE(INET6, 1, ["Have IPv6 support"]))
+
+#
+# Disable Multilink support
+AC_ARG_ENABLE([multilink],
+    AS_HELP_STRING([--disable-multilink], [Disable multilink support]))
+AM_CONDITIONAL(WITH_MULTILINK, test "x${enable_multilink}" != "xno")
+AM_COND_IF([WITH_MULTILINK],
+    AC_DEFINE([HAVE_MULTILINK], 1, ["Have multilink support"]))
+AS_IF([test "x${build_sunos}" = "xyes" && test "x${enable_multilink}" != "xno"],
+    [AC_MSG_ERROR([Multilink is not supported on SunOS, disable using --disable-multilink])])
+
+#
+# Multilink require Trivial Database Support
+AM_CONDITIONAL(WITH_TDB, test "x${enable_multilink}" != "xno")
+AM_COND_IF([WITH_TDB],
+    AC_DEFINE([USE_TDB], 1, ["Include TDB support"]))
+
+#
+# Disable support for activating pppd on packet activity (filter support)
+AC_ARG_ENABLE([filter],
```

Comment:
Is the libpcap library detection section adapted to cross-compilation ? I'm really not used to autotools, sorry if my question is stupid !

Reply:
You are probably right. I'll change it to use --with-pcap and you'd want to specify the folder. 

It's just that I am not going to be able to update the change before I have a computer in front of me, which would be next Friday

Reply:
No worry, --disable-filter is working for now. I'm able to build the whole PPP, but not to install it. I will try a quick fix for installation, if that does not work I might disable this step, so your pull request should build. Later steps would be enabling filters support, then maybe installation to Buildroot target directory (but this is not really useful in this CI).

Reply:
I can't remember but --prefix="path" is one, but I think there is a magic variable "DESTDIR". I know that is different than the previous INSTROOT="" 

Use Google, autotools are pretty standard. I don't remember from the top of my head if DESTDIR is the one. 

You'd do something similar to 

make install DESTDIR={PATH}

it would install it into DESTDIR+prefix

Reply:
Did you have any success with this?

Reply:
Installation seems to work. However, I get problems with cross-compilation : the generated objects on MIPS64 build are x86_64 ones, I'm working on it.

Reply:
I'm quite close to something working (with filters support), but I'm hitting a PPP musl issue when linking pppd :
```
/home/ar/Documents/ppp/buildroot/output/host/lib/gcc/mips64-buildroot-linux-musl/9.3.0/../../../../mips64-buildroot-linux-musl/bin/ld: pppd-session.o: in function `session_start':
session.c:(.text+0xd4): undefined reference to `logwtmp'
/home/ar/Documents/ppp/buildroot/output/host/lib/gcc/mips64-buildroot-linux-musl/9.3.0/../../../../mips64-buildroot-linux-musl/bin/ld: session.c:(.text+0xe4): undefined reference to `logwtmp'
/home/ar/Documents/ppp/buildroot/output/host/lib/gcc/mips64-buildroot-linux-musl/9.3.0/../../../../mips64-buildroot-linux-musl/bin/ld: pppd-session.o: in function `session_end':
session.c:(.text+0x26c): undefined reference to `logwtmp'
/home/ar/Documents/ppp/buildroot/output/host/lib/gcc/mips64-buildroot-linux-musl/9.3.0/../../../../mips64-buildroot-linux-musl/bin/ld: session.c:(.text+0x278): undefined reference to `logwtmp'
collect2: error: ld returned 1 exit status
```
Previously there was a custom hack to detect whether `logwtmp` was present, is it possible to do such a thing with autotools ? https://github.com/ppp-project/ppp/blob/301230f8fde94b5b80e9dfba51816ede340e6b64/pppd/Makefile.linux#L86

Reply:
Absolutely!

That's the entire point of Autotools. With the particular libc variant, do you know if logwtmp is present, or it needs to link with a different library? 

If this function is not present, there should be a function in sys-linux.c that is keyed off whether utmp.h exist on that system. It maybe exist but it may not provide logwtmp

Reply:
Say, 

When I get to make these changes, what's the best way for me to test them? I don't see much in the .yaml file for script, etc...

Reply:
Musl does not provide `logwtmp` as it is deprecated, but some other functions from `libutil` are here, that's why this is a bit tricky. The best way to fix that would be to rewrite the needed PPP parts to use the correct functions, but this is out of the scope of this pull request.
You are right, a custom `logwtmp` is provided in `sys-linux.c`. I will try this evening if not defining `HAVE_LOGWTMP` is enough to make musl libc work.
I can give you the commands to run if you want to test. Maybe it's time to think to a way to make the Buildroot CI available to developers, through a script or a container.
Current working build for glibc and uClibc (with libpcap) is here : https://github.com/RICCIARDI-Adrien/ppp/tree/update_ci_for_autotools

Reply:
@RICCIARDI-Adrien - Could you give it another spin. I did improve the detection of -lutil / logwtmp, if it doesn't find it, it will use the built-in version instead. Also, fixed the --with-pcap instead of using --enable/disable-filter.

Reply:
Build started with your latest changes, fingers crossed !

Reply:
Congrats @enaess, Buildroot CI seems to work like a charm with all libc and libpcap support. However, something went wrong on Solaris with libpcap support (libpcap-related autotools tests seem to be triggered on Solaris whereas the feature is not supported) : https://github.com/RICCIARDI-Adrien/ppp/runs/3088795322?check_suite_focus=true#step:3:2002

Reply:
You can fix the Solaris issue on this PR branch as Solaris CI has not changed.

Reply:
I did fix it, it will compile ... But the buildroot CI isn't updated on my end yet. How can I incorporate your changes here?

Reply:
I've rebased with your changes, build is starting. If it succeeds, I can create a PR for your branch `enaess:ppp-autotools`, so by merging my PR the commit should come to this PR, is that OK for you ?
Another question : are the `Makefile.linux` or `Makefile.sol2` (present for instance in `pppd` directory) still needed ? When CI is working and merged in your PR branch, maybe you can remove all traces of the legacy build system, so end maintainers can update their PPP packages in an easier way ?

Reply:
Thank you and am looking forward to the PR. Yes, I will remove the legacy stuff in a separate change

Reply:
@RICCIARDI-Adrien I did merge your change, thank you so much for all your help here.